tl/candidate_generation/utility.py

```diff
@@ -60,15 +71,16 @@ def create_cfd_canonical(self, df, relevant_columns, column, size, properties, m
         candidates_format = list()
         all_candidates_aux_dict = {}
 
-        for i, row in df.iterrows():
+        def call_es(row):
```

Comment:
same for this method. The code needs to be restructured so that only one threaded function is implemented and called in both canonical and candidates method